src/main/java/com/oodj/vaccspace/controllers/register/RegisterController.java

```diff
@@ -55,13 +55,56 @@ void onBackToLoginPressed(ActionEvent event) {
 
     @FXML
     void onRegisterPressed(ActionEvent event) {
+        //Empty Fields Validation
+        if (vm.getName().equals("") || vm.getPassword().equals("") || vm.getRepeatPassword().equals("") || vm.getPhoneNumber().equals("") || vm.getEmail().equals("") || vm.getIdentificationNumber().equals("")) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Empty fields", "All fields must be filled in!");
+            return;
+        }
+
+        // Name Validation
+        if (vm.getName().matches(".*\\d.*")) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Invalid Name", "The name should not contain any numbers!");
+            return;
+        }
+
+        //Password Validation
+        if (!vm.getPassword().equals(vm.getRepeatPassword())) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Invalid Password", "The passwords do not match!");
+            return;
+        }
+
+        if (vm.getPassword().length() < 8) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Invalid Password", "Your password must have at least 8 characters!");
+            return;
+        }
+
+        //Phone Validation
+        if (!vm.getPhoneNumber().matches("^[0-9]*$")) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Invalid Phone", "Phone number must only have numbers!");
+            return;
+        }
+
+        //Identification Number Validation
+        if (!vm.getIdentificationNumber().matches("^[0-9]*$")) {
+            Page.showDialog(container.getScene().getWindow(), DialogType.ERROR, "Error: Invalid IC Number", "Identification number must only have numbers!");
+            return;
+        }
+
+        //Email Validation
+        if (!vm.getEmail().matches("^[A-Za-z0-9+_.-]+@(.+)$")) {
```

Comment:
This regex matches `xyz@mail`, which is kinda wrong, please make it match `xyz@mail.com`

Reply:
Changes have been made, not sure if the PR will update themselves but yea lmk if i need to create another PR

Reply:
looks good to me